### DISCUSSION TITLE
Prevent duplicate with unique fields value

### DISCUSSION THREAD

**Discussion by yudyananda** [directus/directus#3958] [Feature Requests]
Hi folks, Let say i have collection of posts which has two fields named title and slug, in this case i'm curious about how to prevent duplicate post with the same value from both fields.

**Comment by benhaynes** [comment]
We don't have a manageable unique constraint in Directus yet, but you might be able to set one directly in the database. Otherwise, you could create a custom interface that checks other values for uniqueness... That could be a stop gap solution.

**Reply by yudyananda** [reply]
Too bad. I think this is important feature and will really help content creators to avoid duplicates, which can be redundancy and bad practice in SEO. Well I hope this feature will be available in the near future. 

Meanwhile, can you elaborate about how to create custom interface that checks other uniqueness value?

**Reply by philleepflorence** [reply]
@yudyananda - If you do not have direct access to the DB or if you want to replicate across many instances, these can help:

Create a custom endpoint that can interact with the DB - [Custom API Endpoints](https://docs.directus.io/guides/api-endpoints/)
Create a unique constraint using the D9 DB Library - [Knex - Schema Unique](http://knexjs.org/#Schema-unique)
